### ISSUE TITLE
Build script not working on fedora

### ISSUE THREAD

**Issue by jsmailes** [alus-it/AirspaceConverter#113]
Fedora has slightly different package names for its dependencies (`libzip-devel`, `boost-devel`, `qt-devel`) and a different name for the `qmake` executable (`qmake-qt4`). I think both of these problems could be fixed by switching from install scripts and Makefiles to a `cmake`-based build/installation pipeline, allowing the local machine to handle distribution-specific quirks.

I'm also getting the following error when attempting to run the `airspaceconverter` or `airspaceconverter-gui` executable:
> ./airspaceconverter: error while loading shared libraries: libairspaceconverter.so: cannot open shared object file: No such file or directory
I think this is likely another issue with the build script, but I've yet to figure out where the problem lies.

**Comment by efa** [comment]
OK, we need a tester on Fedora.
Does not exist an executable 'qmake' and neither a link to it?
We can easily modify the script to handle the Fedora distribution, while switching to cmake based build/installation can be little harder. Maybe we can add it in parallel and evaluate how it work.
About the linker error i had to check a thing

**Comment by efa** [comment]
OK, we need a tester on Fedora.
Does not exist an executable 'qmake' and neither a link to it?
We can easily modify the script to handle the Fedora distribution, while switching to cmake based build/installation can be little harder. Maybe we can add it in parallel and evaluate how it work.
About the linker error i had to check a thing

**Comment by efa** [comment]
OK, we need a tester on Fedora.
Does not exist an executable 'qmake' and neither a link to it?
We can easily modify the script to handle the Fedora distribution, while switching to cmake based build/installation can be little harder. Maybe we can add it in parallel and evaluate how it work.
About the linker error i had to check a thing

**Comment by efa** [comment]
do you have the files:
/etc/os-release
/etc/lsb-release
/etc/redhat-release
if so, can you please post the content here

Same fot the output of:
$ lsb_release -si
$ lsb_release -sr
$ uname -s
$ uname -r

**Comment by jsmailes** [comment]
I don't have qmake on my system (`which qmake` returns no result) but I do have qmake-qt4:
```
$ which qmake-qt4
/usr/bin/qmake-qt4
```

I don't have any executable on $PATH called qtchooser.

```
$ cat /etc/os-release
NAME=Fedora
VERSION="31 (Thirty One)"
ID=fedora
VERSION_ID=31
VERSION_CODENAME=""
PLATFORM_ID="platform:f31"
PRETTY_NAME="Fedora 31 (Thirty One)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:31"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f31/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=31
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=31
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
```
```
$ cat /etc/redhat-release
Fedora release 31 (Thirty One)
```
```
$ lsb_release -si
Fedora
```
```
$ lsb_release -sr
31
```
```
$ uname -s
Linux
```
```
$ uname -r
5.6.11-200.fc31.x86_64
```

Hope this helps, let me know if there's anything else I can do! I'm happy to test builds on my machine.

**Comment by efa** [comment]
At first I want to support RedHat/Fedora fixing the script 'build.sh', ongoing.

Then I want to add support for RPM generation with a script like 'debianize.sh' (here we need your help as I do not know RPM generation rules), or generating cross-distro package like AppImage:
https://github.com/alus-it/AirspaceConverter/issues/117

**Comment by jsmailes** [comment]
Fixing build.sh to support a wider variety of systems would certainly be a good thing, although I think if you're planning to deploy on multiple distros or operating systems then it might be worth changing up your build system. CMake has an extension called CPack which abstracts away the process of building packages for different operating systems, it might be worth looking into that in order to generate `.rpm`, `.deb` etc from the same configuration file.

Here are some potentially useful docs I found on using CPack as part of a `CMakeLists.txt`:
- https://cmake.org/cmake/help/latest/module/CPack.html
- https://manpages.debian.org/unstable/cmake-data/cpack-generators.7.en.html
- https://schneide.blog/2013/02/11/build-a-rpm-package-using-cmake/
- https://gitlab.kitware.com/cmake/community/-/wikis/doc/cpack/Packaging-With-CPack

**Comment by efa** [comment]
thank you for docs links, we will evaluate adding Cmake build chain.

Can you please test the RedHat branch of my forked repo:
https://github.com/efa/AirspaceConverter/tree/RedHat
it has a simple patch in 'build.sh' for supporting Fedora with qmake-qt4
I understood make of CLI and SO work, and it is the QT build that fail, so the fix is on that.
Please report any error you get

**Comment by efa** [comment]
about error:
`./airspaceconverter: error while loading shared libraries`
try my fork branch (waiting integration in main repo):
https://github.com/efa/AirspaceConverter/tree/RPATH
tracked in:
https://github.com/alus-it/AirspaceConverter/issues/119

**Comment by efa** [comment]
RPATH branch is now merged to upstream, try to download the sources, and build with the script:
https://github.com/efa/AirspaceConverter/blob/RedHat/build.sh
report any error here.
If binary are produced, executables should find .so in its directory

**Comment by efa** [comment]
now also 'build.sh' is merged with upstream.
Try to download the current master sources, and build them

**Comment by alus-** [comment]
It would be nice to know if now we can close this issue...

**Comment by efa** [comment]
for cmake I created a dedicated issue:
https://github.com/alus-it/AirspaceConverter/issues/128

**Comment by jsmailes** [comment]
Apologies for the delay in getting back on this, thanks for looking into sorting it out.

I've pulled the latest code from master branch and `build.sh` now runs without any error, but get the same "error while loading shared libraries" message when I try to run it.

I managed to fix this error by copying `Release/libairspaceconverter.so` to `/usr/lib` and running `ldconfig` to reload shared libraries. This seems to be done by `make install` or `install.sh`, but it might be good to try to put together a build path that allows this tool to be used without root.

**Comment by jsmailes** [comment]
I just discovered putting `Release/libairspaceconverter.so` in the `buildQt` directory allows the binary to run without any installation - consider copying the shared library there during the build process?

**Comment by efa** [comment]
OK, thank you for test.
good build.sh now work, so generation is OK.

Please move discussion about the .so error in the issue:
https://github.com/alus-it/AirspaceConverter/issues/119

**Comment by alus-** [comment]
Let's close this one if other improvements are required for Fedora we will open new issue(s) for the next releases.
